build_tools/style.fish

```diff
@@ -76,19 +76,19 @@ if set -q fish_files[1]
 end
 
 if set -q python_files[1]
-    if not type -q black
+    if not type -q ruff
         echo
-        echo $yellow'Please install `black` to style python'$normal
+        echo $yellow'Please install `ruff` to style python'$normal
         exit 127
     end
-    echo === Running "$green"black"$normal"
+    echo === Running "$green"ruff format"$normal"
     if set -l -q _flag_check
-        if not black --check $python_files
+        if not ruff format --check $python_files
             echo $red"Python files are not formatted correctly."$normal
             exit 1
         end
     else
-        black $python_files
+        ruff format $python_files
```

Comment:
FWIW, as mentioned in [Clippy in CI](https://github.com/fish-shell/fish-shell/discussions/11584#discussioncomment-13672983),
we have some unpinned packages from PyPI.

- ruff
- sphinx
- sphinx-markdown-builder
- pexpect
- (not in master today) python3-polib, python3-git, python-feedgen
- (fish-site) jekyll
- (fish-site) jekyll-feed

It's probably not a huge deal but if we ever
- run into version mismatch problems,
- or want reproducible builds

then we could specify versions.

Probably using `pyproject.toml`/`uv.lock` (e.g. `uv init && uv add --dev ruff`)

But we can defer that for now since we never had problems with unpinned black.
`sphinx-markdown-builder` is something that we should probably pin because it's not even on PyPI yet.